### ISSUE TITLE
Dependency issue with last version of scoverage

### ISSUE THREAD

**Issue by mickaelmagniez** [playframework/playframework#11461]
Hi

I have an issue while trying to use version 2.0.5 of scoverage on a "bare" play framework project : 

First i create a play project from seed with 
```
sbt new playframework/play-scala-seed.g8
```

And then i add `scoverage` to plugins, so i have 
```scala
addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.17")
addSbtPlugin("org.scoverage" % "sbt-scoverage" % "2.0.5")
```

```scala
name := """play-scala-seed"""
organization := "com.example"
version := "1.0-SNAPSHOT"
scalaVersion := "2.13.8"
lazy val root = (project in file(".")).enablePlugins(PlayScala)
libraryDependencies += guice
libraryDependencies += "org.scalatestplus.play" %% "scalatestplus-play" % "5.0.0" % Test
```

And then trying to run, i have 
```bash
$ sbt coverage

[error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error] 
[error]         * org.scala-lang.modules:scala-xml_2.12:2.1.0 (early-semver) is selected over {1.2.0, 1.1.1, 1.0.6}
[error]             +- org.scoverage:scalac-scoverage-reporter_2.12:2.0.7 (depends on 2.1.0)
[error]             +- org.scala-lang:scala-compiler:2.12.16              (depends on 1.0.6)
[error]             +- com.typesafe.sbt:sbt-native-packager:1.5.2 (scalaVersion=2.12, sbtVersion=1.0) (depends on 1.1.1)
[error]             +- com.typesafe.play:twirl-api_2.12:1.5.1             (depends on 1.2.0)
```

Seems version 2.0 of scoverage is incompatible with scala 2.12, but my application is 2.13

Thank you for your help.

Best regards

**Comment by oalyman** [comment]
Here is a solution for your problem: https://github.com/scala/bug/issues/12632

**Comment by mkurz** [comment]
There will likely more issues come up like this.
What happened is that some plugins (like [sbt-native-packager](https://github.com/sbt/sbt-native-packager/pull/1514)) started to upgrade to scala-xml 2.x. Then the Scala compiler [followed](https://github.com/scala/bug/issues/12632) in its 2.12.17 release. sbt 1.8.x (unreleased yet) [will upgrade](https://github.com/sbt/sbt/pull/7021) to 2.12.17 as well. The current Play 2.8.x branch and it's dependencies still are on scala-xml 1.x at this point. We probably have to think about to also upgrade to 2.x, however I would at least wait for sbt 1.8 to arrive.
Actually its not really an issue, since AFAIK scala-xml 1 and 2 should be compatible, however it's annoying. The error actually can be suppressed by putting this in your `project/plugins.sbt`:
```sbt
ThisBuild / libraryDependencySchemes ++= Seq(
  "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
)
```

**Comment by mkurz** [comment]
This got fixed in Play 2.9/3.0 because all dependencies and sbt-plugins use scala-xml 2.x now.
